src/test/java/com/google/neighborgood/UserInfoServletTest.java

```diff
@@ -0,0 +1,422 @@
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableMap;
+import java.io.*;
+import java.time.Duration;
+import java.time.Instant;
+import javax.servlet.http.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit test on the UserInfoServlet file */
+@RunWith(JUnit4.class)
+public final class UserInfoServletTest {
+
+  /* Set up the test environment with Datastore and UserService and simulate the situation
+   * that the user is logged in as admin, has a specific email and userID.
+   */
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("leonardzhang@google.com")
+          .setEnvAuthDomain("1234567890")
+          .setEnvAttributes(
+              ImmutableMap.of(
+                  "com.google.appengine.api.users.UserService.user_id_key", "1234567890"));
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void setUpTest() {
+    // Test the UserService feature
+    UserService userService = UserServiceFactory.getUserService();
+    assertTrue(userService.isUserAdmin());
+    assertTrue(userService.isUserLoggedIn());
+
+    // Test the DataStore feature
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    assertEquals(0, ds.prepare(new Query("dummy")).countEntities(withLimit(10)));
+    ds.put(new Entity("dummy"));
+    ds.put(new Entity("dummy"));
+    assertEquals(2, ds.prepare(new Query("dummy")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void normalSingleInputTest() {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    // Check whether the datastore is empty before the test
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    assertEquals(0, ds.prepare(new Query("UserInfo")).countEntities(withLimit(10)));
+
+    when(request.getParameter("nickname-input")).thenReturn("Leonard");
+    when(request.getParameter("address-input")).thenReturn("4xxx Centre Avenue");
+    when(request.getParameter("phone-input")).thenReturn("4xxxxxxxxx");
+
+    try {
+      new UserInfoServlet().doPost(request, response);
+    } catch (IOException e) {
+      assertTrue(false);
+    }
```

Comment:
In your tests, I think any exception raised will automatically stop the test and cause it to fail, so we don't need to try-catch these `doPost` calls. We can let the test framework take care of failing the test when there's an exception we're not expecting.

Reply:
Okay. I will remove all these try-catch clauses and declared that the functions may throw IOException in the new commit.